### PR TITLE
refactor(state_table): Cow for get row, minor refactors

### PR DESCRIPTION
**Pull request by skyzh**
Signed-off-by: Alex Chi <iskyzh@gmail.com>

## What's changed and what's your intention?

<del>TopN test doesn't pass, guess there's something wrong with it. Will fix later.</del>

## Checklist

- [x] I have written necessary docs and comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

### CODE REVIEW

File: src/storage/src/table/state_table.rs [hunk 1]
```diff
@@ -67,52 +72,68 @@ impl<S: StateStore> StateTable<S> {
...
+    // TODO: remove, should not be exposed to user
     pub fn get_mem_table(&self) -> &MemTable {
         &self.mem_table
     }
 
-    /// read methods
-    pub async fn get_row(&self, pk: &Row, epoch: u64) -> StorageResult<Option<Row>> {
+    /// Get a single row from state table. This function will return a Cow. If the value is from
+    /// memtable, it will be a [`Cow::Borrowed`]. If is from cell based table, it will be an owned
+    /// value. To convert `Option<Cow<Row>>` to `Option<Row>`, just call `to_owned`.
```
Comment (BugenZhao) [inline src/storage/src/table/state_table.rs:92]:
```suggestion
    /// value. To convert `Option<Cow<Row>>` to `Option<Row>`, just call `into_owned`.
```

File: src/storage/src/table/cell_based_table.rs [hunk 2]
```diff
@@ -57,8 +57,7 @@ pub struct CellBasedTable<S: StateStore> {
     column_descs: Vec<ColumnDesc>,
 
     /// Mapping from column id to column index
-    /// TODO: all CellBasedTable users should have a pk serializer, we should remove this option.
-    pub pk_serializer: Option<OrderedRowSerializer>,
```
Comment (BugenZhao) [inline src/storage/src/table/cell_based_table.rs:61]: Can we remove the `Option` now?

File: src/storage/src/table/state_table.rs [hunk 3]
```diff
@@ -167,7 +188,7 @@ impl<S: StateStore> StateTable<S> {
         self.iter_with_pk_bounds::<_, Row>(.., epoch).await
     }
 
-    /// This function scans rows from the relational table with specific `pk_bounds`.
+    /// This function scans rows from the relational table with specific `pk_prefix`.
```
Comment (BugenZhao) [inline src/storage/src/table/state_table.rs:191]:
```suggestion
    /// This function scans rows from the relational table with specific `pk_bounds`.
```

File: src/storage/src/table/state_table.rs [hunk 4]
```diff
@@ -239,76 +260,71 @@ where
         let cell_based_table_iter = self.cell_based_table_iter.peekable();
```
Comment (BugenZhao) [inline src/storage/src/table/state_table.rs:260]: Also fuse this stream?
```suggestion
        let cell_based_table_iter = self.cell_based_table_iter.fuse().peekable();
```

**Review by BowenXiao1999 — APPROVED** [review]
LGTM

**Review by wcy-fdu — APPROVED** [review]
LSTM

File: src/storage/src/table/state_table.rs (hunk 4 above)
Comment (skyzh) [reply src/storage/src/table/state_table.rs:260]: +1

**Pull request merged**